### DISCUSSION TITLE
Does `_document.mdx` work?

### DISCUSSION THREAD

**Discussion by realfish** [shuding/nextra#2194] [Q&A] [answered]
When I try to add a [custom document](https://nextjs.org/docs/pages/building-your-application/routing/custom-app) with the `.mdx` extension,

```mdx
// _document.mdx
import { Html, Head, Main, NextScript } from 'next/document'
 
export default function Document() {
  return (
    <Html lang="en">
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}
```

I encounter the error below,

```bash
NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted
```

Is the `.mdx` extension available for the custom document feature (`_document`) just as the `_app.mdx`?

A related issue: #1634.

**Comment by dimaMachina** [comment, marked as answer]
do not use _document.mdx, use regular jsx or tsx
